frontend/src/environments/environment.deployment.ts

```diff
@@ -1,7 +1,8 @@
+// https://pumpingco.de/blog/environment-variables-angular-docker/ also see frontend/src/index.html
 import type { Environment } from './environment-type';
 
 export const environment: Environment = {
     production: true,
-    httpPort: 80,
-    websocketPort: 80,
+    httpPort: ((window as any).env.httpPort as number) || 80,
+    websocketPort: ((window as any).env.websocketPort as number) || 80,
```

Comment:
It could be better to use a JSON file instead of a function and load the values from this file (thanks to @Dassderdie for the suggestion).